lib/OpenQA/Schema/ResultSet/Assets.pm

```diff
@@ -38,15 +38,18 @@ sub register {
         }
         return;
     }
-    my $asset = $self->find_or_create(
-        {
-            type => $type,
-            name => $name,
-        },
-        {
-            key => 'assets_type_name',
+
+    return $self->result_source->schema->txn_do(
```

Comment:
I think txn_do will throw once the transaction failed I'm just a bit unsure of the return here... without caching the error at all...

Reply:
I suppose the error handling remains unaffected by that change. We did not handle exceptions before and the new code doesn't either. So in the error case we get log messages like in the [related issue](https://progress.opensuse.org/issues/35749). Is it that bad? What would I do when catching an error? Printing a more fancy log message?

Reply:
Or do you mean it would also throw in case the database is already busy? Because that's actually we error we want to avoid. But I thought txn_do would just block.

Reply:
as I wrote in the ticket: postgresql has a ON CONFLICT DO NOTHING - this avoids any error reporting. It's especially annoying the database log

Reply:
By wrapping the `find_or_create` inside a transaction we prevent the conflict caused by race conditions in advance so no errors regarding this will be logged. I think that's sufficient for what we want. Not sure how to use `ON CONFLICT DO NOTHING` in DBIx. Likely I would have to write the SQL myself for that. I don't think it is worth the effort.

Reply:
I see